File: 1-iniciante/1049/1049.lua
a = io.read()
b = io.read()
c = io.read()

if a == "vertebrado" then		--vertebrado
	if b == "ave" then					--vertebrado, ave
		if c == "carnivoro" then print ("aguia")		--vertebrado, ave, carnivoro
		elseif c == "onivoro" then print ("pomba")		--vertebrado, ave, onivoro
		end
	elseif b == "mamifero" then			--vertebrado, mamifero
		if c == "onivoro" then print ("homem")			--vertebrado, mamifero, onivoro
		elseif c == "herbivoro" then print ("vaca")		--vertebrado, mamifero, herbivoro
		end
	end

elseif a == "invertebrado" then	--invertebrado
	if b == "inseto" then				--invertebrado, inseto
		if c == "hematofago" then print ("pulga")		--invertebrado, inseto, hematofago
		elseif c == "herbivoro" then print ("lagarta")	--invertebrado, inseto, herbivoro
		end
	elseif b == "anelideo" then			--invertebrado, anelideo
		if c == "hematofago" then print ("sanguessuga")	--invertebrado, anelideo, hematofago
		elseif c == "onivoro" then print ("minhoca")	--invertebrado, anelideo, onivoro
		end
	end
end

File: 1-iniciante/1049/1049.cs
using System;

class URI {
    static void Main(string[] args) { 
        string tipo1, tipo2, tipo3;
        tipo1 = Console.ReadLine();
        tipo2 = Console.ReadLine();
        tipo3 = Console.ReadLine();

        if (tipo1 == "vertebrado") {//vertebrado
            if (tipo2 == "ave") {       //vertebrado, ave
                if (tipo3 == "carnivoro") Console.WriteLine("aguia");       //vertebrado, ave, carnivoro
                else Console.WriteLine("pomba");                            //vertebrado, ave, onivoro
            } else {                    //vertebrado, mamifero
                if (tipo3 == "onivoro") Console.WriteLine("homem");         //vertebrado, mamifero, onivoro
                else Console.WriteLine("vaca");                             //vertebrado, mamifero, herbivoro
            }
        } else {                    //invertebrado
            if (tipo2 == "inseto") {    //invertebrado, inseto
                if (tipo3 == "hematofago") Console.WriteLine("pulga");      //invertebrado, inseto, hematofago
                else Console.WriteLine("lagarta");                          //invertebrado, inseto, herbivoro
            } else {                    //invertebrado, anelideo
                if (tipo3 == "hematofago") Console.WriteLine("sanguessuga");//invertebrado, anelideo, hematofago
                else Console.WriteLine("minhoca");                          //invertebrado, anelideo, onivoro
            }
        }
    }
}